pkg/kubelet/kubelet_network.go

```diff
@@ -121,6 +122,64 @@ func (kl *Kubelet) parseResolvConf(reader io.Reader) (nameservers []string, sear
 	return parseResolvConf(reader, scrubber)
 }
 
+func appendAndContinue(composedSearch []string, domain string) bool {
+	searchLen := len(strings.Join(composedSearch, " "))
+
+	if searchLen == 0 || (searchLen+len(domain)+1 <= types.MaxSearchLineLen && len(composedSearch) < types.MaxSearchLineDNSDomains) {
+		return true
+	}
+	return false
+}
+
+func composeDNSSearch(dnsSearch []string, hostSearch []string) []string {
+	composedSearch := []string{}
+
+	//add dns domains with checking  search line limits
+	for _, dnsDomain := range dnsSearch {
+		if appendAndContinue(composedSearch, dnsDomain) {
+			composedSearch = append(composedSearch, dnsDomain)
+		} else {
+			glog.V(8).Infof("Search Line limits were exceeded, some dns names have been omitted, the applied search line is: %s", strings.Join(composedSearch, " "))
+			return composedSearch
+		}
+	}
+
+	//add host domains excluding duplicates
+	for _, hostDomain := range hostSearch {
+		unique := true
+		for _, dnsDomain := range dnsSearch {
+			if hostDomain == dnsDomain {
+				unique = false
+				break
+			}
+		}
+		if !unique {
+			glog.V(8).Infof("Found and omitted duplicated dns domain in host search line: '%s' during merging with cluster dns domains", hostDomain)
+			continue
+		}
+		if appendAndContinue(composedSearch, hostDomain) {
```

Comment:
capture L160-L165 as a function, since it is duplicated above.

```
composedSearch, err := appendOrError(composedSearch, hostDomain)
if err != nil {
  glog...
  return
```